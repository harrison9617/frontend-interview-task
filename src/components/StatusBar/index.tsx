import React from 'react'

type Props = {
  percent: number
  text?: string
}

const StatusBar = ({ percent = 0, text = '' }: Props) => {
  return (
    <div className="">
      <div className="flex items-center justify-between text-sm font-medium mb-1">
        <p className="text-gray-500">{percent.toFixed()}% utilised</p>
        <p className="text-gray-300">{text}</p>
      </div>
      <div className="relative w-full h-1 bg-gray-200 rounded-xl">
        <div
          className={`absolute top-0 left-0 h-1 bg-lime-400 rounded-xl`}
          style={{ width: `${percent}%` }}
        ></div>
      </div>
    </div>
  )
}

export default StatusBar
