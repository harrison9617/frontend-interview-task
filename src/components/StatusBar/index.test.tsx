import { render, waitFor } from '@testing-library/react'
import StatusBar from '@/components/StatusBar'
import '@testing-library/jest-dom'
import 'jest-fetch-mock'

describe('StatusBar', () => {
  it('renders as expected', async () => {
    const { container } = render(<StatusBar percent={0} />)
    expect(container).toMatchSnapshot()
  })
  it('percent should be a number between 0 and 100', async () => {
    const { getByTestId } = render(<StatusBar percent={0} />)
    console.log(getByTestId('percent'))
    expect(Number(getByTestId('percent').textContent)).toBeGreaterThanOrEqual(0);
    expect(Number(getByTestId('percent').textContent)).toBeLessThan(100);
  })
})