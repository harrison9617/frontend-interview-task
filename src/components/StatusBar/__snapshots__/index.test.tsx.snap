// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar renders as expected 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="flex items-center justify-between text-sm font-medium mb-1"
    >
      <p
        class="text-gray-500"
      >
        <span
          data-testid="percent"
        >
          0
        </span>
        % utilised
      </p>
      <p
        class="text-gray-300"
      />
    </div>
    <div
      class="relative w-full h-1 bg-gray-200 rounded-xl"
    >
      <div
        class="absolute top-0 left-0 h-1 bg-lime-400 rounded-xl"
        style="width: 0%;"
      />
    </div>
  </div>
</div>
`;
